packages/components/src/spinner/index.js

```diff
@@ -1,23 +1,47 @@
+/**
+ * External dependencies
+ */
+import classNames from 'classnames';
+
 /**
  * Internal dependencies
  */
-import { StyledSpinner, SpinnerIndicator } from './styles';
+import { StyledSpinner, SpinnerTrack, SpinnerIndicator } from './styles';
 
-export default function Spinner() {
+/**
+ * @typedef OwnProps
+ *
+ * @property {string} [className] Class name
+ */
+/** @typedef {import('react').ComponentPropsWithoutRef<'svg'> & OwnProps} Props */
+
+/**
+ * @param {Props} props
+ * @return {JSX.Element} Element
+ */
+export default function Spinner( { className, ...props } ) {
```

Comment:
Typically we forward refs for all components. I guess this is useful for integrating with a library like Reakit where passing refs for important components happens on a regular basis.

Any reason not to foward the ref for Spinner as well?

Reply:
No reason in particular. I guess this PR was initially focused on updating the visual styles of Spinner, and so we didn't focus as much on other aspects (like forwarding refs, rewriting in typescript, using context etcc)

But it's a quick change and it won't hurt to add it to this PR

Reply:
Oh yeah, good point 🙂 Feel free to do in another PR!

If we do forward the refs then we should be able to use `WordPressComponentProps` with it as well.

Might as well do that in a separate PR as it'll require a few other lines to change as well (including calling `useContext` on props).